packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -295,6 +297,14 @@ export default class ConfigParser {
         // Remove any duplicate tests from the final specs array
         specs = [...new Set(specs)]
 
+        // If the --multi-run flag is set, duplicate the specs array
+        // Ensure that when --multi-run is set that either --spec or --suite is also set
+        if (multiRun && (isSpecParamPassed || suites.length > 0)) {
+            specs = specs.flatMap(i => Array.from({ length: multiRun }).fill(i)) as Spec[]
+        } else if (multiRun && !isSpecParamPassed && suites.length === 0) {
```

Comment:
I think this will be easier to read:
```suggestion
        const hasSubsetOfSpecsDefined = isSpecParamPassed || suites.length > 0
        if (multiRun && hasSubsetOfSpecsDefined) {
            specs = specs.flatMap(i => Array.from({ length: multiRun }).fill(i)) as Spec[]
        } else if (multiRun && !hasSubsetOfSpecsDefined) {
```